opentok/opentok-meet
# [OPENTOK-43154] Lockdown meet.tokbox.com

#### What is this PR doing?

This PR adds google auth to meet.tokbox.com so that only vonage users can create sessions. Once a room/session has been created though any user authenticated/non-authenticated can join. However after 30 minutes the session will be destroyed so any users joining after 30 minutes will need to be authenticated.

#### How should this be manually tested?
- We have 4 new env variables. Please contact me for a config file and I will provide these to you. After setting these run `npm run build` and then `npm start`. Go to http://localhost:3000 like normal.
- When trying to access the root of the page / you should be redirected to google's auth where only logging in with a vonage account will work.
- After logging in with your vonage account you should be redirected back to the page you came from which in this case will be the root.
- Enter a meeting name and hit join.
- You should now be able to invite any user to this call. Try joining using another browser or incognito session where your vonage google user isn't signed in. You should be able to join.
- After 30 minutes the session should be destroyed. So if you try joining from a guest account (non-vonage session) it should not work and send you back to the google sign in page.


#### What are the relevant tickets?
https://jira.vonage.com/browse/OPENTOK-43154
